timemachine/lib/potentials.py

```diff
@@ -1,324 +0,0 @@
-
-    def set_nblist_padding(self, padding):
-        for impl in self.get_potentials():
-            if hasattr(impl, "set_nblist_padding"):
-                impl.set_nblist_padding(padding)
-
-
-class Nonbonded(CustomOpWrapper):
-    def __init__(self, *args):
-
-        # exclusion_idxs should be unique
-        exclusion_idxs = args[1]
-        exclusion_set = set()
-
-        for src, dst in exclusion_idxs:
-            src, dst = sorted((src, dst))
-            exclusion_set.add((src, dst))
-
-        assert len(exclusion_set) == exclusion_idxs.shape[0]
-
-        super(Nonbonded, self).__init__(*args)
-
-    def unbound_impl(self, precision):
-        all_pairs_impl = NonbondedAllPairs(
-            self.get_num_atoms(),
-            self.get_beta(),
-            self.get_cutoff(),
-        ).unbound_impl(precision)
-
-        exclusions_impl = NonbondedPairListNegated(
-            self.get_exclusion_idxs(),
-            self.get_scale_factors(),
-            self.get_beta(),
-            self.get_cutoff(),
-        ).unbound_impl(precision)
-
-        return NonbondedImplWrapper([all_pairs_impl, exclusions_impl])
-
-    def get_num_atoms(self):
-        return self.args[0]
-
-    def set_exclusion_idxs(self, x):
-        self.args[1] = x
-
-    def get_exclusion_idxs(self):
-        return self.args[1]
-
-    def set_scale_factors(self, x):
-        self.args[2] = x
-
-    def get_scale_factors(self):
-        return self.args[2]
-
-    def get_beta(self):
-        return self.args[3]
-
-    def get_cutoff(self):
-        return self.args[4]
-
-
-class NonbondedAllPairs(CustomOpWrapper):
-    def get_num_atoms(self):
-        return self.args[0]
-
-    def get_beta(self):
-        return self.args[1]
-
-    def get_cutoff(self):
-        return self.args[2]
-
-    def get_atom_idxs(self):
-        return self.args[3]
-
-
-class NonbondedInteractionGroup(CustomOpWrapper):
-    def get_num_atoms(self):
-        return self.args[0]
-
-    def get_row_atom_idxs(self):
-        return self.args[1]
-
-    def get_beta(self):
-        return self.args[2]
-
-    def get_cutoff(self):
-        return self.args[3]
-
-
-class NonbondedPairList(CustomOpWrapper):
-    def get_idxs(self):
-        return self.args[0]
-
-    def get_rescale_mask(self):
-        return self.args[1]
-
-    def get_beta(self):
-        return self.args[2]
-
-    def get_cutoff(self):
-        return self.args[3]
-
-
-class NonbondedPairListPrecomputed(CustomOpWrapper):
-    """
-    This implements a pairlist with precomputed parameters. It differs from the regular NonbondedPairlist in that it
-    expects params of the form s0*q_ij, s_ij, s1*e_ij, and w_offsets_ij, where s are the scaling factors and combining
-    rules have already been applied.
-
-    Note that you should not use this class to implement exclusions (that are later cancelled out by AllPairs)
-    since the floating point operations are different in python vs C++.
```

Comment:
Possibly lossy step: I didn't immediately see if the content of this docstring was retained in the code that supersedes this.

Reply:
Good catch. I'll go back over this file and move over any relevant comments.

Reply:
Restored docstring in the new location in https://github.com/proteneer/timemachine/pull/995/commits/b74413f7f885973f1eb87c26fb84589e2d903105